plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineBundle.kt

```diff
@@ -87,6 +93,80 @@ open class GenerateGramineBundle @Inject constructor(
             // The .manifest is not needed for debug and release modes
             outputDir.file(GRAMINE_MANIFEST).get().asFile.toPath().deleteExisting()
         }
+
+        if (!pythonFile.isPresent) {
+            deleteJavaSystemFiles()
+        }
+    }
+
+    private fun deleteJavaSystemFiles() {
+        listOf(JLINK_CUSTOM_JDK_DIRECTORY, SYSTEM_LIB_DIRECTORY).forEach { it ->
+            outputDir.get().asFile.toPath().resolve(it).toFile().deleteRecursively()
+        }
+    }
+
+    private fun prepareBundleForJavaEnclaves(manifestTemplatePath: Path) {
+        val jlinkOutputPath = outputDir.get().asFile.toPath().resolve(JLINK_CUSTOM_JDK_DIRECTORY)
+
+        if (jlinkOutputPath.exists()) {
+            jlinkOutputPath.toFile().deleteRecursively()
+        }
+        createCustomJDK(jlinkOutputPath)
+
+        val outputSystemDir = outputDir.get().asFile.toPath().resolve(SYSTEM_LIB_DIRECTORY)
+
+        if (outputSystemDir.exists()) {
+            outputSystemDir.toFile().deleteRecursively()
+        }
+        copyAdditionalSystemFiles(outputSystemDir)
+        generateManifestForJavaEnclaves(manifestTemplatePath)
+        compressFiles(jlinkOutputPath, outputSystemDir)
+    }
+
+    private fun compressFiles(jlinkOutputPath: Path, outputSystemDir: Path) {
+        createTarFile("$jlinkOutputPath")
+        createTarFile("$outputSystemDir")
+    }
+
+    private fun addFilesToTarGZ(filePath: String, parent: String, tarArchive: TarArchiveOutputStream) {
+        val file = File(filePath)
+        val entryName = parent + file.name
+        tarArchive.putArchiveEntry(TarArchiveEntry(file, entryName))
+        if (file.isFile) {
+            val fis = FileInputStream(file)
+            val bis = BufferedInputStream(fis)
+
+            IOUtils.copy(bis, tarArchive)
+            tarArchive.closeArchiveEntry()
+            bis.close()
+        } else if (file.isDirectory) {
+            tarArchive.closeArchiveEntry()
+
+            if (file.listFiles() != null) {
+                for (f in file.listFiles()!!) {
+                    addFilesToTarGZ(f.absolutePath, entryName + File.separator, tarArchive)
+                }
+            }
+        }
+    }
+
+    private fun createTarFile(sourceDir: String) {
+        var tarOs: TarArchiveOutputStream? = null
+        try {
+            val source = File(sourceDir)
+            val fos = FileOutputStream(source.absolutePath + ".tar.gz")
+            val gos = GZIPOutputStream(BufferedOutputStream(fos))
+            tarOs = TarArchiveOutputStream(gos)
+            addFilesToTarGZ(sourceDir, "", tarOs)
+        } catch (e: IOException) {
+            e.printStackTrace()
+        } finally {
+            try {
+                tarOs!!.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
```

Comment:
Delete this exception handling. If there's an IO error then let it propagate as is.

Reply:
Sure